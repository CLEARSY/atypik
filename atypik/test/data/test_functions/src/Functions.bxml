<?xml version="1.0" encoding="UTF-8" ?>
<Machine xmlns='https://www.atelierb.eu/Formats/bxml' version='1.0' name='Functions' type='abstraction' position='true' semantic='true' b0check='false'>
<Abstract_Variables>
<Attr>
<Pos l='8' c='1' s='18'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='9' c='5' s='3'/>
</Attr>
</Id>
<Id value='var2' typref='4'>
<Attr>
<Pos l='9' c='10' s='4'/>
</Attr>
</Id>
<Id value='var3' typref='0'>
<Attr>
<Pos l='9' c='16' s='4'/>
</Attr>
</Id>
<Id value='var4' typref='5'>
<Attr>
<Pos l='9' c='22' s='4'/>
</Attr>
</Id>
<Id value='var5' typref='4'>
<Attr>
<Pos l='9' c='28' s='4'/>
</Attr>
</Id>
<Id value='fun' typref='6'>
<Attr>
<Pos l='9' c='34' s='3'/>
</Attr>
</Id>
</Abstract_Variables>
<Invariant>
<Attr>
<Pos l='11' c='1' s='9'/>
</Attr>
<Nary_Pred op='&amp;'>
<Exp_Comparison op=':'>
<Attr>
<Pos l='12' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='12' c='5' s='3'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='7'>
<Attr>
<Pos l='12' c='11' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='4'>
<Attr>
<Pos l='12' c='20' s='1'/>
</Attr>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='12' c='15' s='4'/>
</Attr>
</Id>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='12' c='22' s='7'/>
</Attr>
</Id>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='13' c='10' s='1'/>
</Attr>
<Id value='var2' typref='4'>
<Attr>
<Pos l='13' c='5' s='4'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='7'>
<Attr>
<Pos l='13' c='12' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='4'>
<Attr>
<Pos l='13' c='21' s='1'/>
</Attr>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='13' c='16' s='4'/>
</Attr>
</Id>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='13' c='23' s='7'/>
</Attr>
</Id>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='14' c='10' s='1'/>
</Attr>
<Id value='var3' typref='0'>
<Attr>
<Pos l='14' c='5' s='4'/>
</Attr>
</Id>
<Id value='BOOL' typref='9'>
<Attr>
<Pos l='14' c='12' s='4'/>
</Attr>
</Id>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='15' c='10' s='1'/>
</Attr>
<Id value='var4' typref='5'>
<Attr>
<Pos l='15' c='5' s='4'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='10'>
<Attr>
<Pos l='15' c='12' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='5'>
<Attr>
<Pos l='15' c='21' s='1'/>
</Attr>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='15' c='16' s='4'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='3'>
<Attr>
<Pos l='15' c='23' s='3'/>
</Attr>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='15' c='27' s='7'/>
</Attr>
</Id>
</Unary_Exp>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='16' c='10' s='1'/>
</Attr>
<Id value='var5' typref='4'>
<Attr>
<Pos l='16' c='5' s='4'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='7'>
<Attr>
<Pos l='16' c='12' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='4'>
<Attr>
<Pos l='16' c='21' s='1'/>
</Attr>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='16' c='16' s='4'/>
</Attr>
</Id>
<Id value='INTEGER' typref='2'>
<Attr>
<Pos l='16' c='23' s='7'/>
</Attr>
</Id>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='17' c='9' s='1'/>
</Attr>
<Id value='fun' typref='6'>
<Attr>
<Pos l='17' c='5' s='3'/>
</Attr>
</Id>
<Unary_Exp op='POW' typref='11'>
<Attr>
<Pos l='17' c='11' s='3'/>
</Attr>
<Binary_Exp op='*s' typref='6'>
<Attr>
<Pos l='17' c='26' s='1'/>
</Attr>
<Binary_Exp op='*s' typref='12'>
<Attr>
<Pos l='17' c='19' s='1'/>
</Attr>
<Id value='INT' typref='2'>
<Attr>
<Pos l='17' c='15' s='3'/>
</Attr>
</Id>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='17' c='21' s='4'/>
</Attr>
</Id>
</Binary_Exp>
<Id value='BOOL' typref='9'>
<Attr>
<Pos l='17' c='28' s='4'/>
</Attr>
</Id>
</Binary_Exp>
</Unary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='18' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='18' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='--&gt;' typref='7'>
<Attr>
<Pos l='18' c='22' s='3'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='18' c='11' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='18' c='12' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='18' c='17' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='18' c='26' s='1'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='18' c='27' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='18' c='30' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='19' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='19' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='+-&gt;' typref='7'>
<Attr>
<Pos l='19' c='27' s='3'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='19' c='11' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='19' c='12' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='19' c='17' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='2.0' typref='13'>
<Attr>
<Pos l='19' c='22' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='19' c='31' s='1'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='19' c='32' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='19' c='35' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='20' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='20' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='&gt;+&gt;' typref='7'>
<Attr>
<Pos l='20' c='27' s='3'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='20' c='11' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='20' c='12' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='20' c='17' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='2.0' typref='13'>
<Attr>
<Pos l='20' c='22' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='20' c='31' s='1'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='20' c='32' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='20' c='35' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='20' c='38' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='21' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='21' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='&gt;-&gt;' typref='7'>
<Attr>
<Pos l='21' c='22' s='3'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='21' c='11' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='21' c='12' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='21' c='17' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='21' c='26' s='1'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='21' c='27' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='21' c='30' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='21' c='33' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='22' c='10' s='1'/>
</Attr>
<Id value='var2' typref='4'>
<Attr>
<Pos l='22' c='5' s='4'/>
</Attr>
</Id>
<Binary_Exp op='+-&gt;&gt;' typref='7'>
<Attr>
<Pos l='22' c='33' s='4'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='22' c='12' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='22' c='13' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='22' c='18' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='2.0' typref='13'>
<Attr>
<Pos l='22' c='23' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='3.0' typref='13'>
<Attr>
<Pos l='22' c='28' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='22' c='38' s='1'/>
</Attr>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='22' c='39' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='22' c='42' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='23' c='10' s='1'/>
</Attr>
<Id value='var2' typref='4'>
<Attr>
<Pos l='23' c='5' s='4'/>
</Attr>
</Id>
<Binary_Exp op='--&gt;&gt;' typref='7'>
<Attr>
<Pos l='23' c='28' s='4'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='23' c='12' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='23' c='13' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='23' c='18' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='2.0' typref='13'>
<Attr>
<Pos l='23' c='23' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='23' c='33' s='1'/>
</Attr>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='23' c='34' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='23' c='37' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='24' c='9' s='1'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='24' c='5' s='3'/>
</Attr>
</Id>
<Binary_Exp op='&gt;-&gt;&gt;' typref='7'>
<Attr>
<Pos l='24' c='22' s='4'/>
</Attr>
<Nary_Exp op='{' typref='8'>
<Attr>
<Pos l='24' c='11' s='1'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='24' c='12' s='3'/>
</Attr>
</Real_Literal>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='24' c='17' s='3'/>
</Attr>
</Real_Literal>
</Nary_Exp>
<Nary_Exp op='{' typref='2'>
<Attr>
<Pos l='24' c='27' s='1'/>
</Attr>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='24' c='28' s='1'/>
</Attr>
</Integer_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='24' c='31' s='1'/>
</Attr>
</Integer_Literal>
</Nary_Exp>
</Binary_Exp>
</Exp_Comparison>
</Nary_Pred>
</Invariant>
<Initialisation>
<Attr>
<Pos l='26' c='1' s='14'/>
</Attr>
<Nary_Sub op='||'>
<Assignement_Sub>
<Attr>
<Pos l='27' c='9' s='2'/>
</Attr>
<Variables>
<Id value='var' typref='4'>
<Attr>
<Pos l='27' c='5' s='3'/>
</Attr>
</Id>
</Variables>
<Values>
<Nary_Exp op='{' typref='4'>
<Attr>
<Pos l='27' c='12' s='1'/>
</Attr>
<Binary_Exp op='|-&gt;' typref='14'>
<Attr>
<Pos l='27' c='18' s='3'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='27' c='14' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='27' c='22' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
<Binary_Exp op='|-&gt;' typref='14'>
<Attr>
<Pos l='27' c='31' s='3'/>
</Attr>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='27' c='27' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='27' c='35' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
</Nary_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='28' c='10' s='2'/>
</Attr>
<Variables>
<Id value='var2' typref='4'>
<Attr>
<Pos l='28' c='5' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Nary_Exp op='{' typref='4'>
<Attr>
<Pos l='28' c='13' s='1'/>
</Attr>
<Binary_Exp op='|-&gt;' typref='14'>
<Attr>
<Pos l='28' c='19' s='3'/>
</Attr>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='28' c='15' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='1' typref='1'>
<Attr>
<Pos l='28' c='23' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
<Binary_Exp op='|-&gt;' typref='14'>
<Attr>
<Pos l='28' c='32' s='3'/>
</Attr>
<Real_Literal value='1.0' typref='13'>
<Attr>
<Pos l='28' c='28' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='28' c='36' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
<Binary_Exp op='|-&gt;' typref='14'>
<Attr>
<Pos l='28' c='45' s='3'/>
</Attr>
<Real_Literal value='2.0' typref='13'>
<Attr>
<Pos l='28' c='41' s='3'/>
</Attr>
</Real_Literal>
<Integer_Literal value='2' typref='1'>
<Attr>
<Pos l='28' c='49' s='1'/>
</Attr>
</Integer_Literal>
</Binary_Exp>
</Nary_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='29' c='9' s='2'/>
</Attr>
<Variables>
<Id value='fun' typref='6'>
<Attr>
<Pos l='29' c='5' s='3'/>
</Attr>
</Id>
</Variables>
<Values>
<Quantified_Exp type='%' typref='6'>
<Attr>
<Pos l='29' c='24' s='1'/>
</Attr>
<Variables>
<Id value='var' typref='1'>
<Attr>
<Pos l='29' c='14' s='3'/>
</Attr>
</Id>
<Id value='var2' typref='13'>
<Attr>
<Pos l='29' c='19' s='4'/>
</Attr>
</Id>
</Variables>
<Pred>
<Nary_Pred op='&amp;'>
<Exp_Comparison op=':'>
<Attr>
<Pos l='29' c='30' s='1'/>
</Attr>
<Id value='var' typref='1'>
<Attr>
<Pos l='29' c='26' s='3'/>
</Attr>
</Id>
<Id value='INT' typref='2'>
<Attr>
<Pos l='29' c='32' s='3'/>
</Attr>
</Id>
</Exp_Comparison>
<Exp_Comparison op=':'>
<Attr>
<Pos l='29' c='43' s='1'/>
</Attr>
<Id value='var2' typref='13'>
<Attr>
<Pos l='29' c='38' s='4'/>
</Attr>
</Id>
<Id value='REAL' typref='8'>
<Attr>
<Pos l='29' c='45' s='4'/>
</Attr>
</Id>
</Exp_Comparison>
</Nary_Pred>
</Pred>
<Body>
<Boolean_Exp typref='0'>
<Attr>
<Pos l='29' c='52' s='4'/>
</Attr>
<Exp_Comparison op='='>
<Attr>
<Pos l='29' c='67' s='1'/>
</Attr>
<Unary_Exp op='real' typref='13'>
<Attr>
<Pos l='29' c='57' s='4'/>
</Attr>
<Id value='var' typref='1'>
<Attr>
<Pos l='29' c='62' s='3'/>
</Attr>
</Id>
</Unary_Exp>
<Id value='var2' typref='13'>
<Attr>
<Pos l='29' c='69' s='4'/>
</Attr>
</Id>
</Exp_Comparison>
</Boolean_Exp>
</Body>
</Quantified_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='30' c='10' s='2'/>
</Attr>
<Variables>
<Id value='var3' typref='0'>
<Attr>
<Pos l='30' c='5' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Binary_Exp op='(' typref='0'>
<Attr>
<Pos l='30' c='16' s='0'/>
</Attr>
<Id value='fun' typref='6'>
<Attr>
<Pos l='30' c='13' s='3'/>
</Attr>
</Id>
<Binary_Exp op='|-&gt;' typref='15'>
<Integer_Literal value='0' typref='1'>
<Attr>
<Pos l='30' c='17' s='1'/>
</Attr>
</Integer_Literal>
<Real_Literal value='0.0' typref='13'>
<Attr>
<Pos l='30' c='20' s='3'/>
</Attr>
</Real_Literal>
</Binary_Exp>
</Binary_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='31' c='10' s='2'/>
</Attr>
<Variables>
<Id value='var4' typref='5'>
<Attr>
<Pos l='31' c='5' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Unary_Exp op='fnc' typref='5'>
<Attr>
<Pos l='31' c='13' s='3'/>
</Attr>
<Id value='var' typref='4'>
<Attr>
<Pos l='31' c='17' s='3'/>
</Attr>
</Id>
</Unary_Exp>
</Values>
</Assignement_Sub>
<Assignement_Sub>
<Attr>
<Pos l='32' c='10' s='2'/>
</Attr>
<Variables>
<Id value='var5' typref='4'>
<Attr>
<Pos l='32' c='5' s='4'/>
</Attr>
</Id>
</Variables>
<Values>
<Unary_Exp op='rel' typref='4'>
<Attr>
<Pos l='32' c='13' s='3'/>
</Attr>
<Id value='var4' typref='5'>
<Attr>
<Pos l='32' c='17' s='4'/>
</Attr>
</Id>
</Unary_Exp>
</Values>
</Assignement_Sub>
</Nary_Sub>
</Initialisation>
<TypeInfos>
<Type id='0'>
<Id value='BOOL'/>
</Type>
<Type id='1'>
<Id value='INTEGER'/>
</Type>
<Type id='2'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Type>
<Type id='3'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='4'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Id value='INTEGER'/>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='5'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='6'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='REAL'/>
</Binary_Exp>
<Id value='BOOL'/>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='7'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Id value='INTEGER'/>
</Binary_Exp>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='8'>
<Unary_Exp op='POW'>
<Id value='REAL'/>
</Unary_Exp>
</Type>
<Type id='9'>
<Unary_Exp op='POW'>
<Id value='BOOL'/>
</Unary_Exp>
</Type>
<Type id='10'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Unary_Exp op='POW'>
<Id value='INTEGER'/>
</Unary_Exp>
</Binary_Exp>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='11'>
<Unary_Exp op='POW'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='REAL'/>
</Binary_Exp>
<Id value='BOOL'/>
</Binary_Exp>
</Unary_Exp>
</Unary_Exp>
</Type>
<Type id='12'>
<Unary_Exp op='POW'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='REAL'/>
</Binary_Exp>
</Unary_Exp>
</Type>
<Type id='13'>
<Id value='REAL'/>
</Type>
<Type id='14'>
<Binary_Exp op='*'>
<Id value='REAL'/>
<Id value='INTEGER'/>
</Binary_Exp>
</Type>
<Type id='15'>
<Binary_Exp op='*'>
<Id value='INTEGER'/>
<Id value='REAL'/>
</Binary_Exp>
</Type>
</TypeInfos>

</Machine>