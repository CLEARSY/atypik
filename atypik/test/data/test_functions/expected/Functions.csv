Expression	Identifier	Line	Column	Type
"rel(var4)"	141	32	13	(POW (PRODUCT REAL INTEGER))
"fun((0) |-> (0.0))"	135	30	16	BOOL
"0.0"	133	30	20	REAL
"% (var, var2).((var : INT) & (var2 : REAL) | bool((real(var) = var2)))"	131	29	24	(POW (PRODUCT (PRODUCT INTEGER REAL) BOOL))
"bool((real(var) = var2))"	130	29	52	BOOL
"real(var)"	129	29	57	REAL
"(1.0) |-> (2)"	122	28	32	(PRODUCT REAL INTEGER)
"(0.0) |-> (1)"	119	28	19	(PRODUCT REAL INTEGER)
"1"	118	28	23	INTEGER
"(1.0) |-> (1)"	115	27	31	(PRODUCT REAL INTEGER)
"1"	114	27	35	INTEGER
"(0.0) |-> (0)"	112	27	18	(PRODUCT REAL INTEGER)
"({0.0,1.0}) >->> ({0,1})"	107	24	22	(POW (POW (PRODUCT REAL INTEGER)))
"1"	105	24	31	INTEGER
"0"	104	24	28	INTEGER
"var2"	128	29	19	REAL
"0.0"	101	24	12	REAL
"{1,2}"	97	23	33	(POW INTEGER)
"2"	96	23	37	INTEGER
"1"	95	23	34	INTEGER
"1.0"	92	23	18	REAL
"(2.0) |-> (2)"	125	28	45	(PRODUCT REAL INTEGER)
"0.0"	91	23	13	REAL
"1.0"	113	27	27	REAL
"({0.0,1.0,2.0,3.0}) +->> ({1,2})"	88	22	33	(POW (POW (PRODUCT REAL INTEGER)))
"{1,2}"	87	22	38	(POW INTEGER)
"2"	86	22	42	INTEGER
"1"	85	22	39	INTEGER
"0"	111	27	22	INTEGER
"{0.0,1.0,2.0}"	94	23	12	(POW REAL)
"({0.0,1.0}) --> ({0,1})"	46	18	22	(POW (POW (PRODUCT REAL INTEGER)))
"{0,1}"	45	18	26	(POW INTEGER)
"1.0"	50	19	17	REAL
"0"	43	18	27	INTEGER
"({0.0,1.0,2.0}) >+> ({0,1,2})"	67	20	27	(POW (POW (PRODUCT REAL INTEGER)))
"2.0"	82	22	23	REAL
"0"	132	30	17	INTEGER
"1.0"	41	18	17	REAL
"POW((REAL) *s (INTEGER))"	11	12	11	(POW (POW (PRODUCT REAL INTEGER)))
"(REAL) *s (INTEGER)"	13	13	21	(POW (PRODUCT REAL INTEGER))
"0.0"	40	18	12	REAL
"2"	124	28	49	INTEGER
"0.0"	110	27	14	REAL
"((INT) *s (REAL)) *s (BOOL)"	35	17	26	(POW (PRODUCT (PRODUCT INTEGER REAL) BOOL))
"(0) |-> (0.0)"	134	-1	-1	(PRODUCT INTEGER REAL)
"{0,1}"	106	24	27	(POW INTEGER)
"INT"	31	17	15	(POW INTEGER)
"var"	0	9	5	(POW (PRODUCT REAL INTEGER))
"({0.0,1.0,2.0}) -->> ({1,2})"	98	23	28	(POW (POW (PRODUCT REAL INTEGER)))
"var4"	3	15	5	(POW (PRODUCT REAL (POW INTEGER)))
"POW((REAL) *s (INTEGER))"	29	16	12	(POW (POW (PRODUCT REAL INTEGER)))
"2"	65	20	38	INTEGER
"0"	53	19	32	INTEGER
"var5"	4	9	28	(POW (PRODUCT REAL INTEGER))
"2.0"	123	28	41	REAL
"1.0"	120	28	28	REAL
"2.0"	51	19	22	REAL
"{0,1,2}"	66	20	31	(POW INTEGER)
"INTEGER"	7	12	22	(POW INTEGER)
"(REAL) *s (INTEGER)"	8	12	20	(POW (PRODUCT REAL INTEGER))
"0.0"	59	20	12	REAL
"2"	75	21	33	INTEGER
"{(0.0) |-> (1),(1.0) |-> (2),(2.0) |-> (2)}"	126	28	13	(POW (PRODUCT REAL INTEGER))
"0.0"	117	28	15	REAL
"fun"	5	17	5	(POW (PRODUCT (PRODUCT INTEGER REAL) BOOL))
"var2"	1	9	10	(POW (PRODUCT REAL INTEGER))
"2.0"	61	20	22	REAL
"POW(((INT) *s (REAL)) *s (BOOL))"	38	17	11	(POW (POW (PRODUCT (PRODUCT INTEGER REAL) BOOL)))
"(REAL) *s (POW(INTEGER))"	21	15	21	(POW (PRODUCT REAL (POW INTEGER)))
"fnc(var)"	138	31	13	(POW (PRODUCT REAL (POW INTEGER)))
"(INT) *s (REAL)"	32	17	19	(POW (PRODUCT INTEGER REAL))
"REAL"	6	12	15	(POW REAL)
"2"	121	28	36	INTEGER
"BOOL"	18	14	12	(POW BOOL)
"POW((REAL) *s (INTEGER))"	16	13	12	(POW (POW (PRODUCT REAL INTEGER)))
"0.0"	49	19	12	REAL
"({0.0,1.0,2.0}) +-> ({0,1})"	56	19	27	(POW (POW (PRODUCT REAL INTEGER)))
"POW(INTEGER)"	19	15	23	(POW (POW INTEGER))
"{0.0,1.0,2.0}"	52	19	11	(POW REAL)
"1"	54	19	35	INTEGER
"{0,1}"	55	19	31	(POW INTEGER)
"(REAL) *s (INTEGER)"	26	16	21	(POW (PRODUCT REAL INTEGER))
"1.0"	60	20	17	REAL
"{0.0,1.0,2.0}"	62	20	11	(POW REAL)
"1.0"	102	24	17	REAL
"2.0"	93	23	23	REAL
"0"	63	20	32	INTEGER
"var"	127	29	14	INTEGER
"1"	64	20	35	INTEGER
"0.0"	70	21	12	REAL
"{0.0,1.0}"	42	18	11	(POW REAL)
"1.0"	71	21	17	REAL
"{(0.0) |-> (0),(1.0) |-> (1)}"	116	27	12	(POW (PRODUCT REAL INTEGER))
"{0.0,1.0}"	72	21	11	(POW REAL)
"1"	44	18	30	INTEGER
"0"	73	21	27	INTEGER
"1"	74	21	30	INTEGER
"{0,1,2}"	76	21	26	(POW INTEGER)
"({0.0,1.0}) >-> ({0,1,2})"	77	21	22	(POW (POW (PRODUCT REAL INTEGER)))
"0.0"	80	22	13	REAL
"POW((REAL) *s (POW(INTEGER)))"	24	15	12	(POW (POW (PRODUCT REAL (POW INTEGER))))
"1.0"	81	22	18	REAL
"var3"	2	14	5	BOOL
"3.0"	83	22	28	REAL
"{0.0,1.0}"	103	24	11	(POW REAL)
"{0.0,1.0,2.0,3.0}"	84	22	12	(POW REAL)
